buy_together_app/tests/test_login.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+from unittest.mock import Mock
+from buy_together_app import views
+
+
+class Test_Login:
+    @pytest.fixture
+    def authenticate_mock(self, monkeypatch):
+        mock = Mock()
+        monkeypatch.setattr(views, views.authenticate.__name__, mock)
+        return mock
+
+    @pytest.fixture
+    def login_mock(self, monkeypatch):
+        mock = Mock()
+        monkeypatch.setattr(views, views.login.__name__, mock)
+        return mock
+
+    @pytest.fixture
+    def redirect_mock(self, monkeypatch):
+        mock = Mock()
+        monkeypatch.setattr(views, views.redirect.__name__, mock)
+        return mock
+
+    @pytest.fixture
+    def messages_mock(self, monkeypatch):
+        mock = Mock()
+        monkeypatch.setattr(views, "messages", mock)
+        return mock
+
+    @pytest.fixture
+    def render_mock(self, monkeypatch):
+        mock = Mock()
+        monkeypatch.setattr(views, views.render.__name__, mock)
+        return mock
+
+    def test_login_success(
+        self, authenticate_mock, login_mock, redirect_mock, messages_mock, render_mock
+    ):
+        request_mock = Mock()
+        request_mock.method = "POST"
+        output = views.log_in_page(request_mock)
+        assert output == redirect_mock.return_value
+        assert authenticate_mock.call_count == 1
+        assert login_mock.call_count == 1
+        assert redirect_mock.call_count == 1
+        assert messages_mock.info.call_count == 0
+        assert render_mock.call_count == 0
+
+    def test_login_not_post(
+        self, authenticate_mock, login_mock, redirect_mock, messages_mock, render_mock
+    ):
+        request_mock = Mock()
+        request_mock.method = "NOT_POST"
+        output = views.log_in_page(request_mock)
+        assert output == render_mock.return_value
+        assert authenticate_mock.call_count == 0
+        assert login_mock.call_count == 0
+        assert redirect_mock.call_count == 0
+        assert messages_mock.info.call_count == 0
+        assert render_mock.call_count == 1
+
+    def test_login_failed_auth(
+        self, authenticate_mock, login_mock, redirect_mock, messages_mock, render_mock
+    ):
+        request_mock = Mock()
+        request_mock.method = "POST"
+        authenticate_mock.return_value = None
+        output = views.log_in_page(request_mock)
+        assert output == render_mock.return_value
```

Comment:
Isn't there a way to check it actually returns the render we expect it to, instead of a meaningless mock?
also we should test we get the `messages.info(request, 'User name OR Password is incorrect')`